workspaces/ui-v2/src/analytics/implementations/publicExampleAnalytics.ts

```diff
@@ -0,0 +1,98 @@
+import { Client } from '@useoptic/cli-client';
+import * as Sentry from '@sentry/react';
+import * as FullStory from '@fullstory/browser';
+import { LogLevel } from '@sentry/types';
+import { AnalyticsStoreProps } from '../AnalyticsStore';
+import Analytics from '@segment/analytics.js-core/build/analytics';
+// @ts-ignore
+import SegmentIntegration from '@segment/analytics.js-integration-segmentio';
+const packageJson = require('../../../package.json');
+const clientId = `local_cli_${packageJson.version}`;
+
+// @ts-ignore
+const segmentAnalytics = new Analytics();
+
+// If the cli-server is reachable, the identity can be established
+// The public example can also be run in different contexts without a cli-server
+// In which case we fallback to a generic id
+const fetchIdentity = async () => {
+  const cliClient = new Client('/api');
+  try {
+    const response = await cliClient.getIdentity();
+    if (response.ok) {
+      const { anonymousId } = await response.json();
+      return anonymousId;
+    } else {
+      throw new Error();
+    }
+  } catch (e) {
+    return 'public_example_anon_id';
+  }
+};
+
+export const getMetadata: (
+  apiNameFetcher: () => Promise<string>
+) => AnalyticsStoreProps['getMetadata'] = (apiNameFetcher) => async () => {
+  const clientAgent = await fetchIdentity();
+  const apiName = await apiNameFetcher();
+  return {
+    clientId,
+    clientAgent,
+    apiName,
+  };
+};
+
+export const initialize: AnalyticsStoreProps['initialize'] = async (
+  metadata,
+  appConfig
+) => {
+  if (!appConfig.analytics.enabled) {
+    return;
+  }
+
+  if (appConfig.analytics.segmentToken) {
```

Comment:
In the future we may or may not want to exclude / not send events here - or at least distinguish between onboarding vs actual usage